Lib9c/Policy/BlockPolicySource.Utils.cs

```diff
@@ -0,0 +1,380 @@
+            return true;
+        }
+
+        internal static Func<BlockChain<NCAction>, Address, long, bool> IsAllowedToMineFactory(
+            Func<BlockChain<NCAction>, long, bool> isPermissionedMiningBlockIndex,
+            Func<BlockChain<NCAction>, Address, long, bool> isPermissionedToMine)
+        {
+            return (blockChain, miner, index) =>
+                IsAllowedToMineRaw(
+                    blockChain, miner, index, isPermissionedMiningBlockIndex, isPermissionedToMine);
+        }
+
+        /// <summary>
+        /// Checks if authorized mining policy applies to given block index.
+        /// </summary>
+        /// <remarks>
+        /// An implementation should be agnostic about other policies affecting the same index.
+        /// Policy overruling between different policies should be handled elsewhere.
+        /// </remarks>
+        internal static bool IsAuthorizedMiningBlockIndex(
+            BlockChain<NCAction> blockChain, long index)
+        {
+            if (GetAuthorizedMinersState(blockChain) is AuthorizedMinersState ams)
+            {
+                return index % ams.Interval == 0 && index <= ams.ValidUntil;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Checks if given miner is allowed to mine a block with given index according to
+        /// authorized mining policy.
+        /// </summary>
+        /// <remarks>
+        /// An implementation should be agnostic about other policies affecting the same index.
+        /// Policy overruling between different policies should be handled elsewhere.
+        /// </remarks>
+        internal static bool IsAuthorizedToMine(
+            BlockChain<NCAction> blockChain, Address miner, long index)
+        {
+            if (IsAuthorizedMiningBlockIndex(blockChain, index))
+            {
+                if (GetAuthorizedMinersState(blockChain) is AuthorizedMinersState ams)
+                {
+                    return ams.Miners.Contains(miner);
+                }
+                else
+                {
+                    throw new ArgumentException(
+                        $"Result of {nameof(GetAuthorizedMinersState)} cannot be null.");
+                }
+            }
+            else
+            {
+                throw new ArgumentException(
+                    $"Result of {nameof(IsAuthorizedMiningBlockIndex)} must be true.");
+            }
+        }
+
+        /// <summary>
+        /// Checks if permissioned mining policy applies to given block index.
+        /// </summary>
+        /// <remarks>
+        /// An implementation should be agnostic about other policies affecting the same index.
+        /// Policy overruling between different policies should be handled elsewhere.
+        /// </remarks>
+        internal static bool IsPermissionedMiningBlockIndexRaw(
+            BlockChain<NCAction> blockChain,
```

Comment:
`blockChain` doesn't seem to be used. Is it a placeholder for matching types with other interfaces?

Reply:
It's purely a placeholder for consistency, a leftover artifact before deciding whether to either have both `AuthorizedMinersState` and `PermissionedMiningPolicy` in the ledger or as a hardcoded policy. Should be removed once `AuthorizedMinersState` is changed to use hardcoded data.